File: 98-validate-binary-search-tree/validate-binary-search-tree.java
class Solution {
    boolean help(TreeNode r, Long min, Long max) {
        if (r == null)
            return true;
        if (r.val <= min || r.val >= max)
            return false;
        return help(r.left, min, (long)r.val) && help(r.right, (long)r.val, max);
    }

    public boolean isValidBST(TreeNode root) {
        return help(root, Long.MIN_VALUE, Long.MAX_VALUE);
    }
}

File: 98-validate-binary-search-tree/validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def inorder(node, elements):
            if not node:
                return
            inorder(node.left, elements)
            elements.append(node.val)
            inorder(node.right, elements)
        elements = []
        inorder(root, elements)
        for i in range(1, len(elements)):
            if elements[i] <= elements[i-1]:
                return False
        return True
        